ambry-messageformat/src/main/java/com/github/ambry/messageformat/BlobStoreHardDelete.java

```diff
@@ -233,7 +233,7 @@ class HardDeleteRecoveryMetadata {
     userMetadataVersion = stream.readShort();
     userMetadataSize = stream.readInt();
     blobRecordVersion = stream.readShort();
-    if (blobRecordVersion == Blob_Version_V2) {
+    if (blobRecordVersion >= Blob_Version_V2) {
       blobType = BlobType.values()[stream.readShort()];
```

Comment:
So this change imply that all future higher versions have to put blobType in the BlobType.values(), while currently we only do that for Blob_Version_V2. Is that a contract we want to obey?

Reply:
That is correct, BlobType is introduced in V2 and stays in V3.  That's why it is V2+.  In the future, if it gets removed in V4, then we'll need to modify the logic.